### PR TITLE
Hybridized compressible solver parameters and updates

### PR DESCRIPTION
**Pull request by thomasgibson**
This fixes the issue (from the Gusto Slack channel) with the hydrostatic mountain test case. The problem was that the default order of elimination in `firedrake.SCPC` attempted to invert a singular matrix due. For the `h_mountain.py` example, parameters have been set to reverse the order. This fixes the issue.

I also made some other minor changes/corrections. Including adding the `unsaturated_bubble.py` test to the hybrid solver tests. It seems that #169 is resolved, as I was able to run the example to completion without error.

### CODE REVIEW

**Comment by thomasgibson** [comment]
Do not merge yet, I'm running certain "problem examples" to completion to make absolutely sure this is working.

**Comment by thomasgibson** [comment]
I believe this is ready now

File: gusto/linear_solvers.py [hunk 1]
```diff
@@ -389,72 +393,77 @@ def V(u):
...
-        else:
-            theta_w = theta
-            thetabar_w = thetabar
+            pi_avg_solver.solve()
 
         # "broken" u, rho, and trace system
-        eqn = (inner(w, (state.h_project(u) - u_in))*dx
-               - beta_cp*div(theta_w*V(w))*pibar*dxp
-               # following does nothing but is preserved in the comments
-               # to remind us why (because V(w) is purely vertical).
```
Comment (jshipton) [inline gusto/linear_solvers.py]: I notice we've lost this comment and the commented out term is now added - do we need to add it if it does nothing or is the comment no longer true?

Comment (thomasgibson) [reply gusto/linear_solvers.py]: Okay I'm reasonably convinced it's not needed. I've gone ahead and updated the branch

**Comment by thomasgibson** [comment]
Good to go?

**Comment by jshipton** [comment]
Looks good to me @thomasgibson! @colinjcotter did you want to comment or shall I merge?

**Comment by thomasgibson** [comment]
@tommbendall said he'd look at this. Once I get approval, I will go ahead and merge

**Comment by colinjcotter** [comment]
merge please!

________________________________

**Pull request merged**